pkg/cmd/secvulnMaven.go

```diff
@@ -5,33 +5,319 @@
 package cmd
 
 import (
+	"encoding/xml"
 	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sort"
+	"strings"
 
 	"github.com/spf13/cobra"
 )
 
 var (
-    secvulnMavenCmd = &cobra.Command{
+	secvulnMavenCmd = &cobra.Command{
 		Use:   "maven",
 		Short: "Generate psuedo maven project for security vulnerability scanning",
 		Long:  "Generate psuedo maven project for security vulnerability scanning",
 		Run:   secvulnMavenExecute,
 	}
 
-	secvulnMavenParentDir     string
-	secvulnMavenPomUrls    *[]string
+	secvulnMavenParentDir string
+	secvulnMavenPomUrls   *[]string
+	secvulnMavenPomRepos  *[]string
+
+	completedProjects []string
+	toDoProjects      []string
 )
 
 func init() {
 	secvulnMavenCmd.PersistentFlags().StringVar(&secvulnMavenParentDir, "parent", "", "Parent project directory")
 	secvulnMavenPomUrls = secvulnMavenCmd.PersistentFlags().StringArray("pom", nil, "Component Pom URLs")
+	secvulnMavenPomRepos = secvulnMavenCmd.PersistentFlags().StringArray("repo", nil, "Repos to look for Poms")
 
 	secvulnMavenCmd.MarkPersistentFlagRequired("parent")
 	secvulnMavenCmd.MarkPersistentFlagRequired("pom")
+	secvulnMavenCmd.MarkPersistentFlagRequired("repo")
 
 	secvulnCmd.AddCommand(secvulnMavenCmd)
 }
 
 func secvulnMavenExecute(cmd *cobra.Command, args []string) {
 	fmt.Printf("Galasa Build - Security Vulnerability Maven - version %v\n", rootCmd.Version)
+
+	for _, pom := range *secvulnMavenPomUrls {
+		startScanningPipeline(pom)
+	}
+
+	updateParent()
+	fmt.Printf("Security scanning project pom.xml updated\n")
+
+}
+
+func startScanningPipeline(mainPomUrl string) {
+
+	fmt.Printf("Starting the scanning pipeline at %v\n", mainPomUrl)
+
+	mainPom, err := readPomFromUrl(mainPomUrl)
+	if err != nil {
+		fmt.Printf("Unable to find the main pom at %v\n", mainPomUrl)
+		panic(err)
+	}
+
+	createPseudoMavenProject(mainPom)
+	fmt.Printf("Pseudo maven project created for: %v\n", mainPom.ArtifactId)
+
+	// Repeat the process for all projects in this dependency chain if the groupId is dev.galasa
+	fmt.Printf("Pseudo maven projects created for dependency chain of %v:\n", mainPom.ArtifactId)
+
+	for _, project := range toDoProjects {
+
+		// If this project has already been processed then continue
+		if checkIfCompleted(project) {
+			continue
+		}
+
+		// Get the groupId of this project from the main pom to use in the url
+		var groupId string
+		if groupId = getGroupId(project, mainPom); groupId == "" {
+			fmt.Printf("Unable to get groupId for artifact %s\n", project)
+			panic(nil)
+		}
+		// Get the version of this project from the main pom to use in the url
+		var version string
+		if version = getVersion(project, mainPom); version == "" {
+			fmt.Printf("Unable to get version for artifact %s\n", project)
+			panic(nil)
+		}
+
+		// Get the current pom for this project to use to create the stripped down pom for the pseudo maven project
+		var currentPom Pom
+		if project == "com.jcraft.jsch" {
```

Comment:
As you can now have multiple repos to search,  you can exclude this check as we will add maven central,  so this is now longer required.

Reply:
Hi Mike, so I think I need to keep this check in, as if the project's artifactId is com.jcraft.jsch, I have to pass the readPomFromRepos method "jsch" as the artifactId and "com.jcraft" as the groupId, instead of "com.jcraft.jsch" and "dev.galasa". The readPomFromRepos method iterates through the repos given in the galasabld command, so it won't find this artifact at our galasadev repo, but when it checks maven central it should find it there, but only if we've kept this check and given the needed artifactId and groupId